lib/destination/schemePunkDestination.js

```diff
@@ -6,21 +6,30 @@ module.exports = function destinationFactory(pluginName, molotovConfig = {overri
   let setUp;
   const molotov = Molotov(molotovConfig);
   if (pluginName) {
-    setUp = Promise.resolve(molotov.getMolotov().resolve().destination[pluginName]); // Ensures we get any superOverrides from config.
+    setUp = molotov.getMolotov().resolve().destination[pluginName]; // Ensures we get any superOverrides from config.
   }
   else {
-    setUp = Promise.resolve(molotov.mixSupers().getMolotov().getSupers().destination);
+    setUp = molotov.mixSupers().getMolotov().getSupers().destination;
   }
 
-  return setUp.then(extendedClass => class schemePunkDestination extends extendedClass {
+  let msg;
+  if (!setUp) {
+    msg = `Destination was called with no plugins but the super did not exist in the supers Object.`; // eslint-disable-line max-len
+    if (pluginName) {
+      msg = `A destination plugin named: ${pluginName}, was called but did not exist in the destination Object.`; // eslint-disable-line max-len
+    }
+    return Promise.reject(new Error(msg));
+  }
+
+  return Promise.resolve(setUp).then(extendedClass => class schemePunkDestination extends extendedClass { // eslint-disable-line max-len
     init(options, transformedValue, scheme, holdOvers, callPath) {
       return super.init(options, transformedValue, scheme, holdOvers, callPath);
     }
     /**
      * Here for mixins.
      */
     writeDestinationTarget() {
-      super.writeDestinationTarget();
+      return super.writeDestinationTarget();
```

Comment:
Actually, if this is the _only_ thing that happens in there, does this even need to exist?

Reply:
possibly?  I think the comment points to why, maybe because of overriding supers? I actually don't know off hand.

Reply:
I think this can be pulled but also I think there is no harm leaving it in for the moment so i am going to leave it an open an issue to test it out. I tested in one project and it was fine but i want to test in another.